sdk/search/Azure.Search.Documents/src/Utilities/SearchExtensions.cs

```diff
@@ -68,12 +69,62 @@ public static string GetSearchServiceName(this Uri endpoint)
         /// Split a collection of strings by commas.
         /// </summary>
         /// <param name="value">The value to split.</param>
+        /// <param name="hasODataFunctions">Whether nested OData functions should be ignored.</param>
         /// <returns>A collection of individual values.</returns>
-        public static IList<string> CommaSplit(string value) =>
-            string.IsNullOrEmpty(value) ?
+        public static IList<string> CommaSplit(string value, bool hasODataFunctions = false)
+        {
+            return string.IsNullOrEmpty(value) ?
                 new List<string>() :
-                // TODO: #10600 - Verify we don't need to worry about escaping
-                new List<string>(value.Split(','));
+                new List<string>(hasODataFunctions ? GetClauses() : value.Split(','));
+
+            // See https://docs.microsoft.com/azure/search/search-query-odata-orderby and
+            // https://docs.microsoft.com/en-us/azure/search/search-query-odata-syntax-reference
+            IEnumerable<string> GetClauses()
+            {
+                int start = 0;          // Start of the current clause
+                int end = 0;            // End of the current clause
+                bool inString = false;  // Whether we're in a literal string
+                int parens = 0;         // Depth of nested parentheses
+                for (; end < value.Length; end++)
```

Comment:
If you take `value` as input to the function, it seems like you could declare it `static` and avoid generating a capture class that probably needs to be allocated.